app/src/main/cpp/skyline/nce.h

```diff
@@ -6,77 +6,32 @@
-        ~NCE();
-
-        /**
-         * @brief The main event loop of the program
-         */
-        void Execute();
-
-        /**
-         * @brief Execute any arbitrary function on a specific child thread
-         * @param call The specific call to execute
-         * @param funcRegs A set of registers to run the function
-         * @param thread The thread to execute the function on
-         */
-        void ExecuteFunction(ThreadCall call, Registers &funcRegs, std::shared_ptr<kernel::type::KThread> &thread);
+        static void SignalHandler(int signal, siginfo *info, ucontext *context, void *oldTls);
 
-        /**
-         * @brief Execute any arbitrary function on the child process
-         * @param call The specific call to execute
-         * @param funcRegs A set of registers to run the function
-         */
-        void ExecuteFunction(ThreadCall call, Registers &funcRegs);
-
-        /**
-         * @brief Waits till a thread is ready to execute commands
-         * @param thread The KThread to wait for initialization
-         */
-        void WaitThreadInit(std::shared_ptr<kernel::type::KThread> &thread);
+        NCE(const DeviceState &state);
 
-        /**
-         * @brief Sets the X0 and X1 registers in a thread and starts it and it's kernel thread
-         * @param entryArg The argument to pass in for the entry function
-         * @param handle The handle of the main thread
-         * @param thread The thread to set the registers and start
-         * @note This function will block forever if the thread has already started
-         */
-        void StartThread(u64 entryArg, u32 handle, std::shared_ptr<kernel::type::KThread> &thread);
+        struct PatchData {
+            size_t size; //!< Size of the .patch section
+            std::vector<size_t> offsets; //!< Offsets in .text of instructions that need to be patched
+        };
 
-        /**
-         * @brief Prints out a trace and the CPU context
-         * @param numHist The amount of previous instructions to print (Can be 0)
-         * @param ctx The ThreadContext of the thread to log
-         */
-        void ThreadTrace(u16 numHist = 10, ThreadContext *ctx = nullptr);
+        static PatchData GetPatchData(const std::vector<u8> &text);
 
         /**
-         * @brief Patches specific parts of the code
-         * @param code A vector with the code to be patched
-         * @param baseAddress The address at which the code is mapped
-         * @param offset The offset of the code block from the base address
+         * @brief Writes the .patch section and mutates the code accordingly
+         * @param patch A pointer to the .patch section which should be exactly patchSize in size and located before the .text section
          */
-        std::vector<u32> PatchCode(std::vector<u8> &code, u64 baseAddress, i64 offset);
+        static void PatchCode(std::vector<u8> &text, u32* patch, size_t patchSize, const std::vector<size_t>& offsets);
```

Comment:
Can you run the formatter, we have a bunch of pointers and references that are on the wrong side?